### ISSUE TITLE
New option for operator-linebreak

### ISSUE THREAD

**Issue by Zamralik** [eslint/eslint#13820]
**What rule do you want to change?**

operator-linebreak

**Does this change cause the rule to produce more or fewer warnings?**

Either

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option `"around"`, or an object `{ "before": boolean, "after": boolean }`

**Please provide some example code that this change will affect:**

```js
if (
    checkSomething1()
    ||
    checkSomething2()
    &&
    checkSomething3()
    &&
    checkSomething4()
)
{
    doSomething();
}
```

**What does the rule currently do for this code?**

It's invalid no matter which option is picked.

**What will the rule do after it's changed?**

Allow this code to be valid.

**Are you willing to submit a pull request to implement this change?**

No, I do not know how ESLint work internally

**Comment by mdjermanovic** [comment]
Hi @Zamralik, thanks for the issue!

I think it isn't common to always enforce linebreaks on both sides. For instance, that option would make a simple `if (a || b)` being invalid.

`"overrides"` also accepts `"ignore"` value, so you can set `overrides: { "||": "ignore", "&&": "ignore" }` as in [this demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG9wZXJhdG9yLWxpbmVicmVhazogW1wiZXJyb3JcIiwgXCJhZnRlclwiLCB7IG92ZXJyaWRlczogeyBcInx8XCI6IFwiaWdub3JlXCIsIFwiJiZcIjogXCJpZ25vcmVcIiB9IH1dICovXG5cbmlmIChcbiAgICBjaGVja1NvbWV0aGluZzEoKVxuICAgIHx8XG4gICAgY2hlY2tTb21ldGhpbmcyKClcbiAgICAmJlxuICAgIGNoZWNrU29tZXRoaW5nMygpXG4gICAgJiZcbiAgICBjaGVja1NvbWV0aGluZzQoKVxuKVxue1xuICAgIGRvU29tZXRoaW5nKCk7XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19).

**Comment by Zamralik** [comment]
Hi @mdjermanovic 
For my understanding, it wouldn't make `if (a || b)` invalid as `linebreak-operator` is only enforced on multiline binary operation.

What I want to achieve is that:

```js
// valid
if (a && b)
{
    doSomething();
}

// valid
if (
    a
    &&
    b
)
{
    doSomething();
}

// error
if (
    a &&
    b
)
{
    doSomething();
}

// error
if (
    a
    && b
)
{
    doSomething();
}
````

**Comment by mdjermanovic** [comment]
Thanks for clarifying! That logic might work.

Still, I believe this style isn't common. Is it recommended by any popular style guides?

Given our new [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), I think this enhancement doesn't qualify, but I'm not opposed to adding the option if other team members think it's an oversight that `operator-linebreak` doesn't support this style.

**Comment by nzakas** [comment]
As @mdjermanovic points out, we are no longer making changes to stylistic rules like this. Your best bet is to create your own version of the rule to enforce what you'd prefer.
